### PR TITLE
deployment / pure-Docker simplifications

### PR DESCRIPTION
**Pull request by slimsag**
This PR includes various changes to simply cluster deployments in general. I have created it with the intent of supporting https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/18 in specific.

> This PR does not need to update the CHANGELOG because no user-facing changes.

### CODE REVIEW

File: cmd/frontend/main.go [hunk 1]
```diff
@@ -1,5 +1,6 @@
 //docker:user sourcegraph
 //docker:env CONFIGURATION_MODE=server
+//docker:env PUBLIC_REPO_REDIRECTS=true
```
Comment (keegancsmith) [inline cmd/frontend/main.go]: Shouldn't we just update the default for this envvar? The only place I believe we want this set to `false` is production, and we explicitly set that for production.

File: cmd/indexer/main.go [hunk 2]
```diff
@@ -1,5 +1,12 @@
 //docker:user sourcegraph
 
+// Postgres defaults for cluster deployments.
+//docker:env PGDATABASE=sg
+//docker:env PGHOST=pgsql
+//docker:env PGPORT=5432
+//docker:env PGSSLMODE=disable
+//docker:env PGUSER=sg
```
Comment (keegancsmith) [inline cmd/indexer/main.go]: whaaaat, indexer speak to pgsql??

File: cmd/searcher/main.go [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 // Defaults for cluster deployments.
 //docker:env CACHE_DIR=/mnt/cache/searcher
+//docker:env SEARCHER_CACHE_SIZE_MB=100000
```
Comment (keegancsmith) [inline cmd/searcher/main.go]: should we not just set this default in code?

Comment (slimsag) [reply cmd/searcher/main.go]: Currently, the default for dev mode is `0`. I don't know whether or not that matters, so it is possible we could simplify this by moving this to the default in code -- but I wanted to make a 1:1 change here and that would mean I would have to set this in two locations: once for dev mode, and once for the default in code (actually the default for cluster deployments); so I opted for this.

File: cmd/indexer/main.go (hunk 2 above)
Comment (slimsag) [reply cmd/indexer/main.go]: I've looked around, and it appears you are correct it does not. Why on earth do we specify these here then? 😆https://github.com/sourcegraph/deploy-sourcegraph/blob/master/base/indexer/indexer.Deployment.yaml#L32-L41

I'll remove

File: cmd/searcher/main.go (hunk 3 above)
Comment (keegancsmith) [reply cmd/searcher/main.go]: dev mode might as well be the same (100gb).

Comment (slimsag) [reply cmd/searcher/main.go]: Ok, cool, I'll change to that then.

**Pull request merged**